source/MaterialXContrib/MaterialXMaya/Plugin.cpp

```diff
@@ -19,73 +19,122 @@
 namespace MaterialXMaya
 {
 
-Plugin& Plugin::instance()
+namespace
 {
-    static Plugin s_instance;
-    return s_instance;
-}
+class SearchPathBuilder
```

Comment:
Used for library and resource search paths, ensures that no duplicates are added to the search path.